.github/workflows/matomo-tests.yml

```diff
@@ -0,0 +1,173 @@
+name: Matomo Tests
+
+on:
+  pull_request:
+
+permissions:
+  actions: read
+  checks: none
+  contents: read
+  deployments: none
+  issues: read
+  packages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+concurrency:
+  group: php-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  PHP:
+    runs-on: ubuntu-20.04
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      redis-sentinel:
+        image: 'bitnami/redis-sentinel:latest'
+        ports:
+          - 26379:26379
+      mysql:
+        image: mysql:5.7
+        ports:
+          - 3306:3306
+        env:
+          MYSQL_DATABASE: matomo_tests
+          MYSQL_ALLOW_EMPTY_PASSWORD: yes
+        options: --tmpfs /var/lib/mysql:rw
+    strategy:
+      fail-fast: false
+      matrix:
+        command: [ 'UnitTests', 'SystemTestsPlugins' ,'SystemTestsCore' , 'IntegrationTestsCore', 'IntegrationTestsPlugins' ]
+        type: [ 'PHP' ]
+        php: [ '7.2', '8.1' ]
+        adapter: [ 'PDO_MYSQL', 'MYSQLI' ]
+        exclude:
+          - php: '8.1'
+            adapter: 'PDO_MYSQL'
+          - php: '7.2'
+            adapter: 'MYSQLI'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          lfs: false
+          submodules: recursive
```

Comment:
We should do the checkout in the github action directly. Passing through the sha/branch/tag should make it possible to use the version, that should be used for testing. This will also be required for testing plugins. We used to test our plugin against various Matomo versions. In most cases, this is the minimum require version as well as the latest 4.x-dev branch